### PR TITLE
Fixing typos

### PR DESCRIPTION
**Pull request by Fourchaux**
Typos found with https://github.com/codespell-project/codespell

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -2865,7 +2865,7 @@ OCaml 4.13.0 (24 September 2021)
   (Xavier Leroy, report by Richard Bornat, review by Stephen Dolan
    and Mark Shinwell)
 
-- #10371: no longer generatd useless `.cds` file when using
+- #10371: no longer generated useless `.cds` file when using
```
Comment (xavierleroy) [inline Changes]: I think "generate" was intended.  You need a verb here.  Or, better: "stop generating".

File: middle_end/flambda/augment_specialised_args.ml [hunk 2]
```diff
@@ -155,7 +155,7 @@ module Processed_what_to_specialise = struct
        needs a "new inner var" and a "new outer var".  However if there
        is already a lifted projection being introduced around the set
        of closures (corresponding to another new specialised argument),
-       we should re-use its "new outer var" to avoid duplication of
+       we should reuse its "new outer var" to avoid duplication of
```
Comment (xavierleroy) [inline middle_end/flambda/augment_specialised_args.ml:158]: "re-use" is not wrong, just a bit fussy and less common than "reuse"

File: typing/typecore.ml [hunk 3]
```diff
@@ -4665,7 +4665,7 @@ and type_function
                     [type_argument] on the cases, and discard the cases'
                     inferred type in favor of the constrained type. (Function
                     cases aren't inferred, so [type_argument] would just call
-                    [type_expect] straightaway, so we do the same here.)
+                    [type_expect] straight away, so we do the same here.)
```
Comment (xavierleroy) [inline typing/typecore.ml]: "straightaway" is perfectly correct, although rather British.  Americans would write "right away", I guess.

**Review by xavierleroy — COMMENTED** [review]
As usual: many good catches, a few unnecessary changes, and one wrong correction.

File: typing/typecore.ml (hunk 3 above)
Comment (Fourchaux) [reply typing/typecore.ml]: Back to straightaway.

File: Changes (hunk 1 above)
Comment (Fourchaux) [reply Changes]: Corrected.

File: middle_end/flambda/augment_specialised_args.ml (hunk 2 above)
Comment (Fourchaux) [reply middle_end/flambda/augment_specialised_args.ml:158]: Let's keep reuse ...

File: stdlib/camlinternalFormat.ml [hunk 4]
```diff
@@ -2067,7 +2067,7 @@ let fmt_ebb_of_string ?legacy_behavior str =
     | Some flag -> flag
     | None -> true
   (*  When this flag is enabled, the format parser tries to behave as
-      the <4.02 implementations, in particular it ignores most benine
+      the <4.02 implementations, in particular it ignores most benign
       nonsensical format. When the flag is disabled, it will reject any
```
Comment (dra27) [inline stdlib/camlinternalFormat.ml]: While we're here:
```suggestion
      nonsensical formats. When the flag is disabled, it will reject any
```

**Review by dra27 — APPROVED** [review]
Looks great, thanks! I suggest one additional correction, and then let's merge.

File: stdlib/camlinternalFormat.ml (hunk 4 above)
Comment (Fourchaux) [reply stdlib/camlinternalFormat.ml]: Done. TY.

**Comment by dra27** [comment]
Thanks!

**Pull request merged**